pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -813,6 +819,33 @@ func (r *ReconcileUserSignup) updateStatusConditions(userSignup *toolchainv1alph
 	return r.client.Status().Update(context.TODO(), userSignup)
 }
 
+func (r *ReconcileUserSignup) sendDeactivatedNotification(logger logr.Logger, userSignup *toolchainv1alpha1.UserSignup) error {
+	notification := &toolchainv1alpha1.Notification{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      userSignup.Status.CompliantUsername + "-deactivated",
+			Namespace: userSignup.Namespace,
+		},
+		Spec: toolchainv1alpha1.NotificationSpec{
+			UserID:   userSignup.Name,
+			Template: notificationtemplates.UserDeactivated.Name,
+		},
+	}
+
+	err := controllerutil.SetControllerReference(userSignup, notification, r.scheme)
+	if err != nil {
+		logger.Error(err, "Failed to set owner reference for deactivation notification")
+		return err
+	}
+
+	if err := r.client.Create(context.TODO(), notification); err != nil {
+		logger.Error(err, "Failed create deactivation notification resource")
```

Comment:
Any tests for this error?

Reply:
Included in https://github.com/codeready-toolchain/host-operator/pull/289/commits/ba8ad1c44e6ef9e9e86ac3c84d5da84be6ef3300